### PR TITLE
UIKit Nav Cleanup

### PR DESCRIPTION
**Pull request by sghoweri**
Long overdue front-end cleanup of the Nav component to be more maintainable, more easily updated in the future and more performant. Not 100% where I want to be with this ideally (maybe 80%+) but definitely puts us in a much better spot than before.

At a high level, this includes one new feature, but most of this is just under the hood refactoring.

### New Feature (Fix?)
- Only allow one top level nav link open at a time while on larger screens / nav displays as a dropdown menu

### Refactor
- Updates us from using Preact v8 to Preact v10 so we can fully take advantage of Fragments, etc
- Breaks the Nav up into smaller, more manageable sub-components
- Significantly reduces the amount of UI logic required (especially with handling the active Nav item)
- Simplifies the CSS to deduplicate and streamline everything; sets us up for eventually allowing end users to officially be able to customize the UI via CSS Custom Properties

I still need to finish up doing cross browser testing on this but early testing seems very promising.

Side Note: I mistakenly branched this off of https://github.com/pattern-lab/patternlab-node/pull/1034 so I'd recommend we merge that down first to avoid merge conflicts + reduce any potential rework!

### CODE REVIEW

**Comment by bmuenzenmeyer** [comment]
@sghoweri are you using this to make progress against #1103 or can it be independently merged?

**Comment by sghoweri** [comment]
Independent — actually started this before #1103!

**Comment by sghoweri** [comment]
Update: there's at least two small CSS fixes that'll be needed here before this is good to go in IE 11 + Firefox, FYI!

**Comment by bmuenzenmeyer** [comment]
@sghoweri just checking in on this - I havent reviewed in quite a while, citing your above comment

**Comment by JosefBredereck** [comment]
> Update: there's at least two small CSS fixes that'll be needed here before this is good to go in IE 11 + Firefox, FYI!

Do you need a review for this topic or is it not finished at the moment?

File: packages/uikit-workshop/src/scripts/components/pl-nav/nav-dropdown.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
+@import '../../../sass/scss/core.scss';
+
+/**
+ * Nav Dropdown
+ * 1) On larger screens, display as dropdowns that
+ *    hangs over the header
+ */
+ .pl-c-nav__dropdown {
```
Comment (Tregonia) [inline packages/uikit-workshop/src/scripts/components/pl-nav/nav-dropdown.scss]: Looks like there is an extra space of indention here

File: packages/uikit-workshop/src/scripts/components/pl-nav/nav-dropdown.scss [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  @media all and (min-width: $pl-bp-med) {
+    position: absolute;
+    top: 100%; /* 1 */
+    left: 0;
+    min-width: 12rem;
+    border-radius: $pl-border-radius-med;
+    border-style: solid;
+    border-width: 1px;
+    box-shadow: 0 2px 5px rgba(0,0,0,.1);
```
Comment (Tregonia) [inline packages/uikit-workshop/src/scripts/components/pl-nav/nav-dropdown.scss]: Is there a benefit to using `rgba(0,0,0,.1)` vs `rgba($pl-color-black,.1)` here?

File: packages/uikit-workshop/src/scripts/components/pl-nav/nav-dropdown.scss [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    z-index: 1;
+    transition: all $pl-animate-quick ease-out;
+
+    .pl-c-body--theme-sidebar & {
+      position: relative;
+    }
+  }
+
+  .pl-c-body--theme-light & {
+    border-color: rgba($pl-color-black, .2);
```
Comment (Tregonia) [inline packages/uikit-workshop/src/scripts/components/pl-nav/nav-dropdown.scss]: Noticed that in [nav-link.scss](https://github.com/pattern-lab/patternlab-node/pull/1102/commits/4af8abfbed681ac4edff8f88aef99f2b5d90f29b#diff-6581b36fcb1402f0c8df623933ca75a2R20), a leading `0` is included on decimals, but they do not include the leading `0` here. 

I recommend these be consistent.

File: packages/uikit-workshop/src/scripts/components/pl-nav/nav-link.scss [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  align-items: center;
+  margin: 0; // remove default button margin in Safari
+  color: inherit;
+  flex-grow: 1;
+  flex-shrink: 1;
+  padding: 0.6rem 0.6rem;
+  flex-basis: calc(100% - 2.5rem);
+  position: relative;
+
+  &:after {
```
Comment (Tregonia) [inline packages/uikit-workshop/src/scripts/components/pl-nav/nav-link.scss]: Pseudo elements are identified with `::`; single `:` are for pseudo classes

File: packages/uikit-workshop/src/scripts/components/pl-nav/nav-link.scss [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    transition: opacity $pl-animate-quick ease-out;
+    content: '';
+    display: block;
+    position: absolute;
+    top: 50%;
+    background-color: currentColor;
+  }
+
+  &:hover {
+    // background-color: transparent !important;
```
Comment (Tregonia) [inline packages/uikit-workshop/src/scripts/components/pl-nav/nav-link.scss]: Should this commented out code remain; perhaps a `@todo` for context?

**Review by JosefBredereck — APPROVED** [review]

**Pull request merged**